### PR TITLE
rewrite the module docstring

### LINKED ISSUE #982: rewrite the `Images` docstring
The current docstring is useless so we need a rewrite of it.

https://github.com/JuliaImages/Images.jl/blob/9796acc8bf1d56d9574c555656fdba362f84a988/src/Images.jl#L206-L236

My idea is to remove the verbose list and only give a general introduction. For instance, modify from https://juliaimages.org/stable/pkgs/.

### PR DESCRIPTION
**Pull request by johnnychen94**
I deliberately separate this into a new PR so we don't get lost in the lengthy discussions in #971 

closes #982

### CODE REVIEW

**Comment by timholy** [comment]
If we're removing the API summary, perhaps we should make sure that all listed `@reexport`ed modules have their own module docstrings. Otherwise there is some loss to this. But I think we could merge this and then fix those.

**Comment by johnnychen94** [comment]
> If we're removing the API summary, perhaps we should make sure that all listed @reexported modules have their own module docstrings. Otherwise there is some loss to this. But I think we could merge this and then fix those.

This seems to be a 1.0 milestone. I plan to have an issue to track all the exported symbols and add necessary docstrings and demo page before 1.0 release.

File: src/Images.jl [hunk 1]
```diff
@@ -101,33 +101,20 @@ export
...
-    - `shepp_logan`
+Images.jl is an "umbrella package" that exports a set of packages which are useful for
+common image processing tasks. Most of these packages are hosted at JuliaImages,
+JuliaArrays, JuliaIO, JuliaGraphics, and JuliaMath.
+
+The purpose of this package is to have an out-of-box experiences for most of the stable
+functionalities. This means when you do `using Images`, you load a lot of packages that
+would otherwise be imported by lengthy using command, e.g., `using ImageCore, ImageShow,
+ImageTransformations, FileIO`. If you care about package loading time, you should probably
+use those small packages and try to composite your own toolbox.
```
Comment (timholy) [inline src/Images.jl:112]:
```suggestion
use those small packages for a narrower toolbox customized to your specific needs.
```

File: src/Images.jl [hunk 2]
```diff
@@ -101,33 +101,20 @@ export
...
+
+The documentation for the JuliaImages ecosystem can be found in https://juliaimages.org, yet
```
Comment (timholy) [inline src/Images.jl:114]:
```suggestion
The documentation for the JuliaImages ecosystem can be found at https://juliaimages.org. Some
```

File: src/Images.jl [hunk 3]
```diff
@@ -101,33 +101,20 @@ export
...
+there still exist some package-specific documentation. For instance, the documentation for
```
Comment (timholy) [inline src/Images.jl:115]:
```suggestion
dependencies have their own documentation. For instance, the documentation for
```

File: src/Images.jl [hunk 4]
```diff
@@ -101,33 +101,20 @@ export
...
-
-Test images and phantoms (see also TestImages.jl):
-
-    - `shepp_logan`
+Images.jl is an "umbrella package" that exports a set of packages which are useful for
+common image processing tasks. Most of these packages are hosted at JuliaImages,
+JuliaArrays, JuliaIO, JuliaGraphics, and JuliaMath.
+
+The purpose of this package is to have an out-of-box experiences for most of the stable
+functionalities. This means when you do `using Images`, you load a lot of packages that
```
[the remark this answers is not in the document]
Comment (johnnychen94) [reply src/Images.jl]: > for most of the stable functionalities.

I do have some concerns with this. I'll write up a new issue to explain it.

[the remark this answers is not in the document]
Comment (johnnychen94) [reply src/Images.jl]: See https://github.com/simonster/Reexport.jl/pull/37. This is why I think we can't expect Images to provide up-to-date functionalities if we want to follow the semver strictly. We have abused it a lot but I think we need to be more cautious on this after v1.0

**Pull request merged**